tests/functional-utils/test_return_value_decorators_type_inference.py

```diff
@@ -0,0 +1,69 @@
+import mypy.api
+from typing import List
+
+import pytest
+
+MYPY_ARGS = ["--ignore-missing-imports"]
+FIXTURE_DIR = "fixtures/mypy/"
+
+
+def fixture_dir(val: str):
+    return FIXTURE_DIR + val
+
+
+def check_mypy_run(
+    cmd_line: List[str],
+    expected_out: str,
+    expected_err: str = "",
+    expected_returncode: int = 1,
+) -> None:
+    """Helper to run mypy and check the output."""
+    out, err, returncode = mypy.api.run(cmd_line)
+    assert out == expected_out, err
+    assert err == expected_err, out
+    assert returncode == expected_returncode, returncode
+
+
+@pytest.mark.parametrize(
+    "fixture,message",
+    (
+        (
+            fixture_dir("to_tuple_decorator.py"),
+            fixture_dir(
+                "to_tuple_decorator.py:13: error: Revealed type is 'builtins.tuple[builtins.int*]'\n"
```

Comment:
It's a little weird that this is an error, but not really an error (just output from `reveal_type()`), and also is something we want. Probably deserves a little explanation.

Reply:
I added a short explanation